### DISCUSSION TITLE
How to change available chains from a child component

### DISCUSSION THREAD

**Discussion by grandsmarquis** [rainbow-me/rainbowkit#1384] [Q&A] [answered]
Not sure if this question fit here as it may more be related to architecture on my side but still trying..

I'm working on fastdapp.xyz
The goal of the web app is to enable users to create their apps (which is a mdx file). They can define a list of chains they allow their user to interact with while using their app.

When I start my app I define the default chains available. But in the component that renders the app, I read the user's config and want to update the rainbow available chains to only the ones the user-defined. 

Here I initialize the wagmi config:
https://github.com/verynifty/fastdapp/blob/main/webapp/pages/_app.js#L39

Here I read the config file but I have no idea how to update the rainbow network selector from here:
https://github.com/verynifty/fastdapp/blob/main/webapp/components/render.js#L131

Any ideas? 

Thanks

**Comment by DanielSinclair** [comment, marked as answer]
You will need to create your own state/provider wrapper for this, but essentially you will need to do two things:
1. Supply all supported chains to `configureChains` and `getDefaultWallets`
2. Pass a subset of chains to `<RainbowKitProvider chains={subsetChains}>`; RainbowKit would re-render when this list changes
